homeassistant/components/mqtt/client.py

```diff
@@ -93,6 +94,55 @@
 
 SubscribePayloadType = str | bytes  # Only bytes if encoding is None
 
+_CallableT = TypeVar("_CallableT", bound=Callable[..., Any])
+
+
+def log_mqtt_callback_exception(func: _CallableT) -> Callable[..., _CallableT]:
```

Comment:
```
    If you wish to suppress exceptions within a callback, you should set
    `client.suppress_exceptions = True`
```

Maybe we should set this instead as we are effectively doing the same thing here

https://github.com/eclipse/paho.mqtt.python/blob/9782ab81fe7ee3a05e74c7f3e1d03d5611ea4be4/src/paho/mqtt/client.py#L482

Reply:
When paho swallows them it still logs

https://github.com/eclipse/paho.mqtt.python/blob/9782ab81fe7ee3a05e74c7f3e1d03d5611ea4be4/src/paho/mqtt/client.py#L2158